docs/basics/gas.md

```diff
@@ -80,6 +85,11 @@ types other than the ones defined by the `evm` module. EVM module transactions c
 as gas is required inherently by the EVM. This check is done by the EVM transactions stateless validation
 (i.e `ValidateBasic`) function as well as on the custom `AnteHandler` defined by Ethermint.
 
+## Gas estimation
+
+Ethereum provides an rpc endpoint `eth_estimateGas` to help users setting up a correct gas limit in their transactions. In Ethermint, a query request is made to the EVM module and the optimal gas limit is evaluated using a binary search on `applyTransaction` to the EVM using a cache context.
```

Comment:
```suggestion
Ethereum provides a JSON-RPC endpoint `eth_estimateGas` to help users set up a correct gas limit in their transactions. In Ethermint, a query request is made to the EVM module and the optimal gas limit is evaluated using a binary search on `applyTransaction` to the EVM using a cache context.
```

Reply:
I think the second sentence is not very clear. From the perspective of someone reading this for the first time without any previous knowledge: Why is binary search used? What is a cache context? how is the transaction applied (is it simulated or do the changes get persisted)? what is the difference with the tx simulation from the SDK that already provides gas estimation and why is it not used here?